testcontainers/src/core/host_port_mapping.rs

```diff
@@ -0,0 +1,169 @@
+use std::collections::BTreeMap;
+
+use crate::core::{client::Client, containers::request::Host, error::TestcontainersError};
+
+/// Checks if we're running on Docker Desktop (macOS or Windows).
+/// Docker Desktop provides `host.docker.internal` by default.
+fn is_docker_desktop() -> bool {
+    let is_desktop = cfg!(any(target_os = "macos", target_os = "windows"));
```

Comment:
This won’t actually work as is because we can’t rely on the operating system of the user for that.

Testcontainers support remote Docker hosts, so the OS you’re running tests on may be different from the remote's one.